t/sendreport/two-tier/email-and-open311.t

```diff
@@ -0,0 +1,379 @@
+use FixMyStreet::Cobrand;
+use FixMyStreet::Script::Reports;
+use FixMyStreet::TestMech;
+use Test::Deep;
+use Test::MockModule;
+use Test::MockObject;
+
+my $mech = FixMyStreet::TestMech->new;
+
+my $user = $mech->create_user_ok('user@example.com');
+
+my $body_oxf = $mech->create_body_ok( 2237, 'Oxfordshire County Council' );
+my $body_cherwell
+    = $mech->create_body_ok( 2419, 'Cherwell District Council' );
+
+$mech->create_contact_ok(
+    body_id  => $body_oxf->id,
+    category => 'Other',
+    email    => 'other@oxfordshire.com',
+);
+$mech->create_contact_ok(
+    body_id  => $body_cherwell->id,
+    category => 'Other',
+    email    => 'other@cherwell.com',
+);
+
+my ($report) = $mech->create_problems_for_body(
+    1,
+    ( join ',', $body_oxf->id, $body_cherwell->id ),
+    'Test',
+    {   cobrand  => 'fixmystreet',
+        category => 'Other',
+        user     => $user,
+    }
+);
+
+$body_oxf->update(
+    {   send_method  => 'Open311',
+        endpoint     => 'http://endpoint.example.com',
+        jurisdiction => 'FMS',
+        api_key      => 'test',
+    },
+);
+
+my $cobrand = FixMyStreet::Cobrand->get_class_for_moniker('fixmystreet')->new;
+$report->result_source->schema->cobrand($cobrand);
+
+my $mock_email   = Test::MockModule->new('FixMyStreet::SendReport::Email');
+my $mock_open311 = Test::MockModule->new('FixMyStreet::SendReport::Open311');
+my $mock_log     = Test::MockObject->new;
+$mock_log->mock( debug => sub { } );
+my $mock_catalyst = Test::MockObject->new;
+$mock_catalyst->mock( log => sub {$mock_log} );
```

Comment:
Is there a way not to need this mock_catalyst thing (and the constant re-setting of it)? As I understand it, this is needed because you're testing `duration_string` - I don't think this testing code needs to worry about that, can it just check `$problem->body`?